src/window/mouse_manager.rs

```diff
@@ -263,16 +256,12 @@ impl MouseManager {
                 modifier_string: keyboard_manager.format_modifier_string(true),
             }
             .into();
-            for _ in 0..(new_y - previous_y).abs() {
+            for _ in 0..(y as i32).abs()+1 {
```

Comment:
And I think the culprit for that is here. Consider the case of a very small movement under `1.0`, which is almost every scroll movement on my system. Previously, those changes accumulated in `self.scroll_position` until they were large enough to cause a difference of `1.0`. But now, every single one of these is reflected and basically `1.0` (because `0.whatever as i32` is `0` and `+ 1`), which means every single frame I'm scrolling has an effect.

I'm not too sure how to correctly handle this though. Maybe we could restore the previous behavior and multiply `y` by a configurable value to accommodate every possible system, since it probably worked fine on yours.

Reply:
Most mouse wheels are *ticked* meaning they are not smooth, so I think that it is fine to call the `nvim_input_mouse()` api function, every tick for mouse input. I think we should have different handling of touch, touchpad and mouse instead.

For touch and touchpad, I think what we could do is to calculate `pixels per line` and when those many pixels are scrolled, we can sent a single `nvim_input_mouse()` api call. But, this is not without caveats!
Under `:h 'mousescroll'` you'll see that a single api call results in different amount of lines scrolled.
The default being `ver:3,hor:6` meaning it'll scroll 3 vertical lines or 6 horizontal lines depending on the parameters passed.

Possible solutions:
1) We scroll by 3 lines every `pixels per line` scroll
2) We scroll by 3 lines every 3*`pixels per line` scroll
3) Enforce a default of `mousescroll=ver:1,hor:1` on neovide users

Reply:
(sorry for the delay, lost track on this one) To be honest, I think option 3 sounds like the best one, even if I usually advocate for not touching user-configurable options at all if possible. I can't figure out a case where scrolling 3 lines at once is more desirable than a flexibly configurable sensitivity config option.